File: cpp/day_26_Sum Root to Leaf Numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void find(TreeNode *root, int curr, int &ans){
        if(root){
            if(!root->left && !root->right){
                ans += curr * 10 + root->val;
                return;
            }
            find(root->left, curr * 10 + root->val, ans);
            find(root->right, curr * 10 + root->val, ans);
        }
    }
    
    int sumNumbers(TreeNode* root) {
        int ans = 0;
        find(root, 0, ans);
        return ans;
    }
};

File: python3/june/day_26_Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def find(self, root, curr):
        if(root is not None):
            if(not root.left and not root.right):
                self.ans += curr * 10 + root.val
                return
            self.find(root.left, curr * 10 + root.val)
            self.find(root.right, curr * 10 + root.val)

    def sumNumbers(self, root: TreeNode) -> int:
        self.ans = 0
        self.find(root, 0)
        return self.ans
